glib/src/subclass/type_plugin.rs

```diff
@@ -0,0 +1,186 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::translate::IntoGlib;
+use crate::translate::{FromGlib, ToGlibPtr};
+use crate::{
+    subclass::prelude::*, Cast, Interface, InterfaceInfo, Type, TypeInfo, TypePlugin,
+    TypeValueTable,
+};
+
+pub trait TypePluginImpl: ObjectImpl + TypePluginImplExt {
+    fn use_plugin(&self) {
+        self.parent_use_plugin();
+    }
+
+    fn unuse_plugin(&self) {
+        self.parent_unuse_plugin();
+    }
+
+    fn complete_type_info(
+        &self,
+        type_: Type,
+        info: &mut TypeInfo,
+        value_table: &mut TypeValueTable,
+    ) {
+        self.parent_complete_type_info(type_, info, value_table);
+    }
+
+    fn complete_interface_info(
+        &self,
+        instance_type: Type,
+        interface_type: Type,
+        info: &mut InterfaceInfo,
+    ) {
+        self.parent_complete_interface_info(instance_type, interface_type, info);
+    }
+}
+
+pub trait TypePluginImplExt: ObjectSubclass {
+    fn parent_use_plugin(&self);
+    fn parent_unuse_plugin(&self);
+    fn parent_complete_type_info(
+        &self,
+        type_: Type,
+        info: &mut TypeInfo,
```

Comment:
Why does this have to be mutable? There's nothing we (can) change in it, is there?

Reply:
that sounds like this vfunc should be returning `(glib::TypeInfo, glib::TypeValueTable)` then

Reply:
My question was more how an implementation in Rust would even create or modify a `TypeInfo` right now. But I see, that makes sense for future extensibility then.

Is the `TypeInfo` passed in by GObject here already correctly initialized, or is the implementer supposed to return a new one? That would basically be what decides if the mutable references or the return tuple is the right choice.